lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -0,0 +1,256 @@
+"""
+Tests for the Tool Meister modules.
+"""
+import io
+import logging
+import pathlib
+import shutil
+import subprocess
+from http import HTTPStatus
+
+import pytest
+import responses
+
+from pbench.agent.tool_meister import ToolMeister, ToolMeisterError, get_logger
+
+tar_file = "test.tar.xz"
+
+
+@pytest.fixture()
+def mock_tar(monkeypatch, agent_setup):
+    def fake_run(*args, **kwargs):
+        assert kwargs["cwd"] == agent_setup["tmp"].parent
+        assert kwargs["stdin"] is None
+        assert kwargs["stderr"] == subprocess.STDOUT
+        assert kwargs["stdout"] == subprocess.PIPE
+        c = subprocess.CompletedProcess(args, returncode=0, stdout=b"", stderr=None)
+        assert all(
+            x in c.args[0]
+            for x in [
+                "tar_path",
+                "--create",
+                "--xz",
+                "--force-local",
+                f"--file={tar_file}",
+            ]
+        )
+        return c
+
+    monkeypatch.setattr(subprocess, "run", fake_run)
+
+
+@pytest.fixture(scope="function")
+def logger_fixture(request):
+    logger = get_logger("__logger__")
```

Comment:
Each logger is global state, so you should consider a unique name for the logger so that they are separaete.

Reply:
The scope of this fixture is a function level so at the end of any individual test all the things defined inside the fixture should be gone isn't it? Why does the logger still remains there for whatever reason even after the fixture is gone

Reply:
Because `get_logger` is from https://github.com/npalaska/pbench/blob/update_tar/lib/pbench/agent/tool_meister.py#L1712, which is just using `logging.getLogger()`, which is always global state (see https://docs.python.org/3.6/library/logging.html#logger-objects: "Multiple calls to `getLogger()` with the same name will always return a reference to the same `Logger` object.").

Also, it is not clear why the unit tests need to use the Tool Meister's `get_logger`.